packages/amplify_auth_cognito/example/ios/amplify_auth_cognito_tests/MockCognito.swift

```diff
@@ -0,0 +1,224 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import Amplify
+import AmplifyPlugins
+import AWSCore
+import AWSPluginsCore
+import AWSMobileClient
+@testable import amplify_auth_cognito
+
+class MockCognito: AuthCognitoProtocol {
+    
+    var returnError: Bool = false
+    
+    func onSignUp(flutterResult: @escaping FlutterResult, request: FlutterSignUpRequest) {
+        if (!returnError) {
+            var signUpRes = Result<AuthSignUpResult,AuthError>.success(
+                AuthSignUpResult(AuthSignUpStep.confirmUser(AuthCodeDeliveryDetails(destination: DeliveryDestination.email(_email)), ["foo": "bar"])))
```

Comment:
Are we sharing these global vars `_email` and `_phoneNumber` with amplify_auth_cognito_tests.swift file? If yes, can we figure out a better way? Such as passing this information in MockCognito constructor to keep it as its instance state for one and only one test?

Reply:
Not relevant anymore.